mrp_workorder_cancel/models/mrp_workorder.py

```diff
@@ -0,0 +1,27 @@
+# Copyright (C) 2023 Open Source Integrators
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import _, models
+from odoo.exceptions import UserError
+
+
+class MrpWorkorder(models.Model):
+    _inherit = "mrp.workorder"
+
+    def button_cancel(self):
+        self.ensure_one()
+        action = self.env["ir.actions.actions"]._for_xml_id(
+            "mrp_workorder_cancel.action_work_order_cancel_confirm"
+        )
+        return action
```

Comment:
FYI this is redundant - a Button can directly call a Window Action, without the need of a model method in the middle.